server/src/lib/config.rs

```diff
@@ -144,6 +148,77 @@ impl BlockDevice {
 
                 Ok((be, child))
             }
+            "crucible" => {
+                let targets: Vec<SocketAddrV4> = self
+                    .options
+                    .get("targets")
+                    .ok_or_else(|| {
+                        ParseError::KeyNotFound(
+                            "targets".to_string(),
+                            "options".to_string(),
+                        )
+                    })?
+                    .as_array()
+                    .ok_or_else(|| {
+                        ParseError::AsError(
+                            "targets".to_string(),
+                            "as_array".to_string(),
+                        )
+                    })?
+                    .to_vec()
+                    .iter()
+                    .map(|x| {
+                        x.as_str()
+                            .ok_or_else(|| {
+                                ParseError::AsError(
+                                    "x".to_string(),
+                                    "as_str".to_string(),
+                                )
+                            })?
+                            .to_string()
```

Comment:
I think you can just omit this (here and a few other places below)? `parse` is defined on `&str` anyways.